torch/distributed/tensor/_sharding_prop.py

```diff
@@ -366,6 +367,41 @@ def propagate_op_sharding_non_cached(self, op_schema: OpSchema) -> OutputShardin
         # special case op, we don't need to propagate for local
         # scalar. TODO: figure out a better way to handle this
         if op_schema.op is aten._local_scalar_dense.default:
+            # there should only be one argument for _local_scalar_dense
+            arg = op_schema.args_schema[0]
+
+            if isinstance(arg, DTensorSpec):
+                has_partial = False
+                for placement in arg.placements:
+                    if isinstance(placement, Partial):
+                        has_partial = True
+                        break
+
+                if has_partial:
+                    # Need to reduce partial values to replicate before calling item()
+                    # Replace Partial placements with Replicate, keep others unchanged
+                    from torch.distributed.tensor.placement_types import Replicate
+
+                    new_placements = tuple(
+                        Replicate() if isinstance(p, Partial) else p
+                        for p in arg.placements
+                    )
+                    replicate_spec = DTensorSpec(
+                        arg.mesh,
+                        new_placements,
+                        arg.tensor_meta,
+                    )
+                    suggestion_schema = OpSchema(
+                        op_schema.op,
+                        (replicate_spec,),
+                        {},
+                    )
+                    suggestion_schema._inplace_rewrap_schema_suggestion(op_schema)
+                    return OutputSharding(
+                        None,
+                        suggestion_schema,
+                        # needs_redistribute=True,
```

Comment:
clean this up? (is this being set to true automatically in the rewrap_schema thing above, when there is a partial?